docs/GUIDE.md

```diff
@@ -0,0 +1,379 @@
+
+Once the 3D avatar is loaded, we'll traverse its scene graph and create an object with all the nodes. This will give us quick access to the `headMesh`. 
+
+```javascript
+// append url parameters to glb url - load ReadyPlayerMe avatar with morphtargets
+const rpmMorphTargetsURL = glbURL + '?morphTargets=ARKit&textureAtlas=1024'
+let nodes
+// Load the GLB with morph targets
+const loader = new GLTFLoader()
+loader.load(rpmMorphTargetsURL, 
+  async (gltf) => {
+  const avatar = gltf.scene
+  // build graph of avatar nodes
+  nodes =  await getGraph(avatar)
+  const headMesh = nodes['Wolf3D_Avatar']
+  // adjust position 
+  avatar.position.y = -1.65
+  avatar.position.z = 1
+  
+  // add avatar to scene
+  scene.add(avatar)
+},
+(event) => {
+  // outout loading details
+  console.log(event)
+})
+```
+
+To account for the noticeable delay between the time the scene is initialized and the moment the 3D avatar is loaded. It's good practice to display a loading animation to inform the user the model is loading and remove it once the 3D avatar is added to the scene.
+
+```javascript
+// show a loading animation
+const loadingDiv = document.createElement('div')
+loadingDiv.classList.add('lds-ripple')
+loadingDiv.append(document.createElement('div'))
+localUserContainer.append(loadingDiv)
+
+/* loader.load - success callback */
+loadingDiv.remove() // remove the loading spinner
+```
+
+## Init video element with Agora
+![](/docs/images/Agora-to-video_element.png "Use Agora Video Track in <video/> element")
+
+We're using Agora to get camera access and create the video and audio tracks. We'll use the camera's video track as the source for the video element. If you'd like a deeper explanation check out my guide on using [Agora with custom video elements](https://medium.com/agora-io/custom-video-elements-with-javascript-and-agora-web-sdk-3c70d5dc1e09).
+
+```javascript
+// Init the local mic and camera
+await initDevices('music_standard', '1080_3')
+// Create video element
+const video = document.createElement('video')
+video.setAttribute('webkit-playsinline', 'webkit-playsinline');
+video.setAttribute('playsinline', 'playsinline');
+// Create a new MediaStream using camera track and set it the video's source object
+video.srcObject = new MediaStream([localMedia.video.track.getMediaStreamTrack()])
+```
+## MediaPipe Setup
+Before we can detect faces and facial gestures, we need to download the latest WASM files for MediaPipe's computer vision and configure a `FaceLandmarker` task. In the face landmarks configuration, we'll set the `faceLandmarker` to output weights for blend shapes (shape keys) and facial transformations. These two settings are important because we'll need access to that data when we run the prediction loop. 
```

Comment:
I am having a hard time understanding this sentence. Might be fine as this is guide is written for someone with a basic understanding of ThreeJS (which I don't have), but maybe it could still be explained a little simpler?

Reply:
Things that I don't understand. What is 
* face landmarks configuration
* output weights for blend shapes (shape keys)
* prediction loop

Reply:
updated this section to add more context.